sound/soc/sof/pm.c

```diff
@@ -220,8 +220,8 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 
 	sof_tear_down_pipelines(sdev, false);
 
-	/* release trace */
-	snd_sof_release_trace(sdev);
+	/* suspend DMA trace */
+	snd_sof_trace_suspend(sdev, SOF_DSP_PM_D3);
```

Comment:
You need to pass the `pm_state`, not a specific state which might not be the right state.